src/boundarySolver/physics/logLawBoundary.cpp

```diff
@@ -0,0 +1,77 @@
+#include "logLawBoundary.hpp"
+#include <utilities/mathUtilities.hpp>
+#include "finiteVolume/compressibleFlowFields.hpp"
+
+void ablate::boundarySolver::physics::LogLawBoundary::Setup(ablate::boundarySolver::BoundarySolver &bSolver) {
+    bSolver.RegisterFunction(UpdateBoundaryVel, this, {finiteVolume::CompressibleFlowFields::EULER_FIELD}, {finiteVolume::CompressibleFlowFields::VELOCITY_FIELD});
+}
+
+PetscErrorCode ablate::boundarySolver::physics::LogLawBoundary::UpdateBoundaryVel(PetscInt dim, const ablate::boundarySolver::BoundarySolver::BoundaryFVFaceGeom *fg,
+                                                                                  const PetscFVCellGeom *boundaryCell, const PetscInt *uOff, PetscScalar *boundaryValues,
+                                                                                  const PetscScalar *stencilValues, const PetscInt *aOff, PetscScalar *auxValues, const PetscScalar *stencilAuxValues,
+                                                                                  void *ctx) {
+    PetscFunctionBeginUser;
+    const PetscInt EULER_FIELD = 0;
+    const PetscInt VEL = 0;
+
+    PetscReal logLawVel[3];
+    PetscArrayzero(logLawVel, 3);
+    PetscReal stencilVel[3];
+    PetscReal stencilNormalVelocity[3];
+    PetscReal theta;
+    PetscReal tangVel;
+    PetscReal tangBoundaryVel;
+
+    // map the stencil velocity in normal coord.
+    PetscReal stencilDensity = stencilValues[uOff[EULER_FIELD] + finiteVolume::CompressibleFlowFields::RHO];
+    for (PetscInt d = 0; d < dim; d++) {
+        stencilVel[d] = stencilValues[uOff[EULER_FIELD] + finiteVolume::CompressibleFlowFields::RHOU + d] / stencilDensity;
+    }
+
+    PetscReal transformationMatrix[dim][3];
+    utilities::MathUtilities::ComputeTransformationMatrix(dim, fg->normal, transformationMatrix);
+    ablate::utilities::MathUtilities::Multiply(dim, transformationMatrix, stencilVel, stencilNormalVelocity);
+    PetscReal area = utilities::MathUtilities::MagVector(dim, fg->areas);
+
+    // calculate the new boundary velocity in normal coord for 2D.
+    logLawVel[0] = 0e+0;
+
+    logLawVel[1] =
```

Comment:
Thanks for making these changes!  We could save a little work by wrapping this in a if dim == 2 else if dim == 3 statement.  Would you mind making this change?

Reply:
Makes sense!